src/utils/DataConfig.py

```diff
@@ -0,0 +1,208 @@
+        raise NotImplementedError(
+            "init subclass of Config that defines the method activity_idx_to_activity_name"
+        )
+
+    # generic
+    def raw_label_to_activity_idx(self, label: str) -> int:
+        """
+        from the label as it is saved in the dataset, to the activity index
+        (Relabeling)
+        """
+        assert (
+            self.raw_label_to_activity_idx_map is not None
+        ), "init subclass of Config that defines the var raw_label_to_activity_idx_map"
+        return self.raw_label_to_activity_idx_map[label]
+
+    def activity_idx_to_activity_name(self, activity_idx: int) -> str:
+        assert (
+            self.activity_idx_to_activity_name_map is not None
+        ), "init subclass of Config that defines the var activity_idx_to_activity_name_map"
+        assert_type((activity_idx, int))
+        return self.activity_idx_to_activity_name_map[activity_idx]
+
+    def n_activities(self) -> int:
+        assert (
+            self.activity_idx_to_activity_name_map is not None
+        ), "init subclass of Config that defines the var activity_idx_to_activity_name_map"
+        return len(self.activity_idx_to_activity_name_map)
+
+
+class OpportunityConfig(DataConfig):
+    def __init__(self, dataset_path: str):
+        self.dataset_path = dataset_path
+
+        self.raw_label_to_activity_idx_map = {
+            0: 0,
+            101: 1,
+            102: 2,
+            103: 3,
+            104: 4,
+            105: 5,
+        }
+
+        self.activity_idx_to_activity_name_map = {
+            0: "null",
+            1: "relaxing",
+            2: "coffee time",
+            3: "early morning",
+            4: "cleanup",
+            5: "sandwich time",
+        }
+
+    def load_dataset(self) -> "list[Recording]":
+        return load_opportunity_dataset(self.dataset_path)
+
+
+class SonarConfig(DataConfig):
+    def __init__(self, dataset_path: str):
+        self.dataset_path = dataset_path
+
+        labels = list(
+            itertools.chain.from_iterable(
+                category["entries"] for category in self.category_labels
+            )
+        )
+        self.raw_label_to_activity_idx_map = {label: i for i, label in enumerate(labels)} # no relabeling applied
+        activities = {k: v for v, k in enumerate(labels)}
+        self.activity_idx_to_activity_name_map = {v: k for k, v in activities.items()}
+
+        # SONAR SPECIFIC VARS --------------------------------------------------
+
+        self.sensor_suffix_order = ["LF", "LW", "ST", "RW", "RF"]
+        self.csv_header_size = 8
+
+    def load_dataset(self, **args) -> "list[Recording]":
+        return load_sonar_dataset(self.dataset_path, **args)
+
+    people = [
+        "unknown",
+        "christine",
+        "aileen",
+        "connie",
+        "yvan",
+        "brueggemann",
+        "jenny",
+        "mathias",
+        "kathi",
+        "anja",
+    ]
+    category_labels = [
+        {
```

Comment:
I think there are labels missing in this list which would lead to us not loading all recordings.